### ISSUE TITLE
Uploading Template Image Throws `Undefined variable:` error in WYSIWYG Templates

### ISSUE THREAD

**Issue by mmilutinovic1313** [elmsln/elmsln#885]
Upon uploading an image the templates throw this error:

`Array
(
    [type] => 8
    [message] => Undefined variable: sidebar_first
    [file] => /var/www/elmsln/core/dslmcode/shared/drupal-7.x/themes/ulmus/adminimal_theme/templates/maintenance-page.tpl.php
    [line] => 27
)
`
![undefinedvariableerror](https://cloud.githubusercontent.com/assets/7243665/15156109/7b5d8d9c-16b3-11e6-9e79-ad330316edd6.png)

**Comment by mmilutinovic1313** [comment]
https://github.com/elmsln/elmsln/issues/785

**Comment by mmilutinovic1313** [comment]
Notes: 

Textbook_WYSIWYG_Template.inc Replace fid's. And drag the images into the template's images folder. Add the name to the load_alter array. That should match the ones on the file system. Bricking is another issue that may be addressed. Maybe test the updated thing.

If there is association -- override the template image reference.

**Comment by btopro** [comment]
https://www.drupal.org/project/wysiwyg_template

**Comment by mmilutinovic1313** [comment]
Use a later dev build not stable

**Comment by btopro** [comment]
See if https://www.drupal.org/node/2503071 helps at all too

**Comment by mmilutinovic1313** [comment]
So I just thought of another approach to this today (it may be the same one we were talking about last time, but I think I simplified for myself to understand it better). 

So the big issue we run into is the fact that when we add new templates we can't upload the image using the upload button. We need the ability to add them to the image folder and update the array on upload. I have uploaded all existing image files to the image folder and updated them in the array, but we don't want to have to do this every time. On upload the array should update and the image should be in the folder automatically.
